src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/processing/advanced/GregtechMetaTileEntity_Adv_DistillationTower.java

```diff
@@ -364,29 +364,15 @@ public String getMachineType() {
     }
 
     @Override
-    public boolean checkRecipe(final ItemStack aStack) {
-        // Run standard recipe handling for distillery recipes
-        if (mMode == Mode.Distillery) {
-            return this.checkRecipeGeneric(getMaxParallelRecipes(), getEuDiscountForParallelism(), 100);
-        } else {
-            ItemStack[] inputs = getCompactedInputs();
-
-            for (GT_MetaTileEntity_Hatch_Input hatch : mInputHatches) {
-                FluidStack tFluid = hatch.getFluid();
-                if (tFluid != null) {
-                    if (checkRecipeGeneric(
-                            inputs,
-                            new FluidStack[] { tFluid },
-                            getMaxParallelRecipes(),
-                            100,
-                            250,
-                            10000)) {
-                        return true;
-                    }
-                }
-            }
-            return false;
-        }
+    protected ProcessingLogic createProcessingLogic() {
+        return new ProcessingLogic().setMaxParallelSupplier(this::getMaxParallelRecipes);
+    }
+
+    @Override
+    protected void setupProcessingLogic(ProcessingLogic logic) {
+        super.setupProcessingLogic(logic);
+        logic.setEuModifier(mMode == Mode.Distillery ? 0.15F : 1F);
+        logic.setEuModifier(mMode == Mode.Distillery ? 1F : 3.5F);
```

Comment:
typo, this should be `setSpeedBonus`. Also aren't speed bonuses `1f/2f` / `1f/3.5f`?

Reply:
1/3.5 yes, the other one just has no speed bonus.

Reply:
`checkRecipeGeneric(getMaxParallelRecipes(), getEuDiscountForParallelism(), 100)` This means `aSpeedBonusPercent == 100` which is 1/2 of duration